src/mod_mam_muc_odbc_arch.erl

```diff
@@ -541,25 +547,36 @@ prepare_filter(RoomID, Borders, Start, End, WithJID, SearchText) ->
                   SWithNick :: escaped_jid() | undefined,
                   SearchText :: binary() | undefined) -> filter().
 make_filter(RoomID, StartID, EndID, SWithNick, SearchText) ->
-   ["WHERE room_id='", escape_room_id(RoomID), "'",
+   ["WHERE room_id=", use_escaped_integer(escape_room_id(RoomID)),
      case StartID of
          undefined -> "";
-         _         -> [" AND id >= ", integer_to_list(StartID)]
+         _         -> [" AND id >= ", use_escaped_integer(escape_integer(StartID))]
      end,
      case EndID of
          undefined -> "";
-         _         -> [" AND id <= ", integer_to_list(EndID)]
+         _         -> [" AND id <= ", use_escaped_integer(escape_integer(EndID))]
      end,
      case SWithNick of
          undefined -> "";
-         _         -> [" AND nick_name = '", SWithNick, "'"]
+         _         -> [" AND nick_name = ", use_escaped_string(SWithNick)]
      end,
      case SearchText of
          undefined -> "";
-         _         -> [" AND search_body like '%", SearchText, "%'"]
+         _         -> prepare_search_filters(SearchText)
      end
     ].
 
+%% Constructs a separate LIKE filter for each word.
+%% SearchText example is "word1%word2%word3".
```

Comment:
it isn't.
It depends how we want to search.

"%word1%word2%" would enforce word order.
like "%word1%" and like "%word2%" would fetch both "word1 word2" and "word2 word1".
I just feel that we want separate likes there. But I can patch it to use strict order.

# Offtop
I still feel that we should implement a separate search backend for elasticsearch :D 
Because rdbms searches are slow and results are not sorted by relevancy.

Reply:
I'm OK with ignoring the order but please leave a note in `mod_mam` docs about it.

Reply:
From docs

> The exact behaviour, like whether word ordering matters, may depend on the storage backend in use.

Reply:
Ah, OK, thanks :)